conftest.py

```diff
@@ -144,19 +144,15 @@ def get_hostlist() -> t.Optional[t.List[str]]:
                 cobalt_fp = os.environ["COBALT_NODEFILE"]
                 with open(cobalt_fp, "r", encoding="utf-8") as nodefile:
                     lines = nodefile.readlines()
-                    test_hostlist = list(
-                        dict.fromkeys([line.strip() for line in lines])
-                    )
+                    test_hostlist = list({line.strip() for line in lines})
             except Exception:
                 return None
         elif "PBS_NODEFILE" in os.environ and not shutil.which("aprun"):
             try:
                 pbs_fp = os.environ["PBS_NODEFILE"]
                 with open(pbs_fp, "r", encoding="utf-8") as nodefile:
                     lines = nodefile.readlines()
-                    test_hostlist = list(
-                        dict.fromkeys([line.strip() for line in lines])
-                    )
+                    test_hostlist = list({line.strip() for line in lines})
```

Comment:
could we pull the 4 lines out into a function (144-147 and 152-155) so the process of de-duping and processing the node files is in one place?

Reply:
Good idea! I'll tack something on this PR rq!!